Code/Framework/AzFramework/AzFramework/Entity/GameEntityContextComponent.cpp

```diff
@@ -340,15 +426,288 @@ namespace AzFramework
+                    break;
+
+                case AZ::Entity::State::Active:
+                    // Deactivate immediately
+                    entity->SetActive(false);
+                    break;
+
+                default:
+                    // Don't do anything, it's not even active.
+                    break;
+                }
+            }
+        }
+    }
+
+    //=========================================================================
+    // GameEntityContextComponent::DeactivateEntityAndDesendants
+    //=========================================================================
+    void GameEntityContextComponent::DeactivateGameEntityAndDescendants(AZ::EntityId rootEntityId, bool updateRoot)
+    {
+        // Verify that this context has the right to perform operations on the entity
+        if (!IsOwnedByThisContext(rootEntityId))
+        {
+            AZ_Warning("EntityContext", false, "Entity %llu not owned by this context.", rootEntityId);
+            return;
+        }
+        
+        AZ::Entity* rootEntity = FindEntity(rootEntityId);
+        if (!rootEntity)
+        {
+            AZ_Warning("EntityContext", false, "Root entity %llu not found.", rootEntityId);
+            return;
+        }
+
+        AZStd::vector<AZ::EntityId> tree;
+        tree.reserve(64);
+        GetEntityTreeFromRootEntity(rootEntityId, tree);
+
+        if(tree.empty()) { return; }
+        
+        // Loop through the vector and disable bottom to top.
+        for (size_t i = tree.size(); i-- > 0; )
+        {
+            AZ::EntityId entityId = tree[i];
+            if(entityId == rootEntityId) 
+            {
+                if(updateRoot) { rootEntity -> SetActive(false); }
+                else { rootEntity->SetActiveByTypeIndex(parentActiveTypeIndex, false); }  
+            }
+            else
+            {
+                AZ::Entity* e = FindEntity(entityId);
+                
+                if(!e) { continue; }
+
+                // Because we're going bottom to top, there's no need to check if the state has changed.
+                e->SetActiveByTypeIndex(parentActiveTypeIndex, false);
+            }
+        }
+    }
+
+    //=========================================================================
+    // EntityContextEventBus::SetGameEntityParent
+    //=========================================================================
+    //Incomplete, needs TransformComponent updates.
+    void GameEntityContextComponent::SetGameEntityParent(const AZ::EntityId& targetEntityId, const AZ::EntityId& parentEntityId)
+    {
+        if(!targetEntityId.IsValid()) { return; }
+
+        AZ::Entity* e = FindEntity(targetEntityId);
+        AZ::Entity* parentE;
+
+        if(!e) { return; }
+
+        //Determine which arrangement of active states is going on.
+        AZ::u32 fromTo = 0; //0 active > Active, 1 Inactive -> Active, 2 Active -> Inactive, 3 Inactive -> Inactive
+
+        if(parentEntityId.IsValid())
```

Comment:
Is this not the wrong way around?

I think the transform component is the source of truth for all parent changes, this would imply that this component needs to listen to the transform bus for such changes, somehow, or have a new message on a new bus, or an event callback, or something it can be told about, rather than trying to change it, itself...